packages/repack/src/webpack/plugins/__tests__/ModuleFederationPlugin.test.ts

```diff
@@ -97,4 +98,109 @@ describe('ModuleFederationPlugin', () => {
     );
     (container.ModuleFederationPlugin as jest.Mock).mockClear();
   });
+
+  it('should add default shared dependencies', () => {
+    new ModuleFederationPlugin({}).apply(jest.fn() as any);
+    const config = (container.ModuleFederationPlugin as jest.Mock).mock
+      .calls[0][0];
+    expect(config.shared).toHaveProperty('react');
+    expect(config.shared).toHaveProperty('react-native');
+    expect(config.shared).toHaveProperty('react-native/');
+    expect(config.shared).toHaveProperty('@react-native/');
+  });
+
+  it('should not add deep imports to defaulted shared dependencies', () => {
+    new ModuleFederationPlugin({ reactNativeDeepImports: false }).apply(
+      jest.fn() as any
+    );
+    const config = (container.ModuleFederationPlugin as jest.Mock).mock
+      .calls[0][0];
+    expect(config.shared).toHaveProperty('react');
+    expect(config.shared).toHaveProperty('react-native');
+    expect(config.shared).not.toHaveProperty('react-native/');
+    expect(config.shared).not.toHaveProperty('@react-native/');
+  });
+
+  it('should add deep imports to existing shared dependencies', () => {
+    new ModuleFederationPlugin({
+      shared: {
+        react: Federated.SHARED_REACT,
+        'react-native': Federated.SHARED_REACT_NATIVE,
+      },
+    }).apply(jest.fn() as any);
+    const config = (container.ModuleFederationPlugin as jest.Mock).mock
+      .calls[0][0];
+    expect(config.shared).toHaveProperty('react-native/');
+    expect(config.shared).toHaveProperty('@react-native/');
```

Comment:
Same assertions are made for `new ModuleFederationPlugin({})` in "should add default shared dependencies" test. How is adding `shared` different?